src/ScriptCs/ExecuteScriptCommandBase.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using ScriptCs.Command;
+using ScriptCs.Contracts;
+
+namespace ScriptCs
+{
+    public abstract class ExecuteScriptCommandBase
+    {
+        protected readonly string _script;
+        protected readonly IFileSystem _fileSystem;
+        protected readonly IScriptExecutor _scriptExecutor;
+        protected readonly IScriptPackResolver _scriptPackResolver;
+        protected readonly ILog _logger;
+        protected readonly IAssemblyResolver _assemblyResolver;
+        protected readonly IFileSystemMigrator _fileSystemMigrator;
+        protected readonly IScriptLibraryComposer _composer;
```

Comment:
I don't think we should be exposing protected fields. It feels like a violation of the open/closed principle.